### PR TITLE
New mechanism to dismiss local notifications just for ios10

### LINKED ISSUE #252: Notifications - iOS10 issues with notifications
- Notifications are not aggregated
  img_0021
- Do not know STR yet, but it is possible to see more than one magnet notification on the lockscreen

### PR DESCRIPTION
**Pull request by arcturus**
Fixes #252

### CODE REVIEW

**Comment by arcturus** [comment]
@wilsonpage r?

File: ios/NotificationsHelper.swift [hunk 1]
```diff
@@ -19,9 +20,15 @@ import UIKit
   class func updateNotifications() {
     guard enabled else { return }
     
-    UIApplication.sharedApplication().cancelAllLocalNotifications()
-    // Clean any badge
-    UIApplication.sharedApplication().applicationIconBadgeNumber = 0
+    if #available(iOS 10.0, *) {
```
Comment (wilsonpage) [inline ios/NotificationsHelper.swift:31]: High level comment to explain why we're doing this.

File: ios/NotificationsHelper.swift [hunk 2]
```diff
@@ -19,9 +20,15 @@ import UIKit
...
+      let center = UNUserNotificationCenter.currentNotificationCenter()
+      center.removeAllDeliveredNotifications()
+    } else {
+      UIApplication.sharedApplication().cancelAllLocalNotifications()
+      // Clean any badge
+      UIApplication.sharedApplication().applicationIconBadgeNumber = 0
+    }
```
Comment (wilsonpage) [inline ios/NotificationsHelper.swift:38]: Pull this block out into a `.clear()` function.

File: ios/NotificationsHelper.swift [hunk 3]
```diff
@@ -19,9 +20,15 @@ import UIKit
...
-    UIApplication.sharedApplication().cancelAllLocalNotifications()
-    // Clean any badge
-    UIApplication.sharedApplication().applicationIconBadgeNumber = 0
+    if #available(iOS 10.0, *) {
+      let center = UNUserNotificationCenter.currentNotificationCenter()
+      center.removeAllDeliveredNotifications()
+    } else {
+      UIApplication.sharedApplication().cancelAllLocalNotifications()
+      // Clean any badge
+      UIApplication.sharedApplication().applicationIconBadgeNumber = 0
```
Comment (wilsonpage) [inline ios/NotificationsHelper.swift:37]: Can we grab this reference once?

```
let app = UIApplication.sharedApplication()
```

**Comment by wilsonpage** [comment]
SHHHHHHHHIII

**Comment by wilsonpage** [comment]
PPPP.LY

**Pull request merged**